hnn_core/optimization/optimizer.py

```diff
@@ -0,0 +1,382 @@
+"""Parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from .metrics import _rmse_evoked
+
+from skopt import gp_minimize
+from scipy.optimize import fmin_cobyla
+
+
+class Optimizer:
+    def __init__(self, net, constraints, set_params, solver, obj_fun,
+                 tstop, scale_factor=1., smooth_window_len=None):
+        """Parameter optimization.
+
+        Parameters
+        ----------
+        net : Network
+            The network object.
+        constraints : dict
+            The user-defined constraints.
+        set_params : func
+            User-defined function that sets parameters in network drives.
+        solver : string
+            The optimizer, 'bayesian' or 'cobyla'.
+        obj_fun : string
+            The type of drive to be optimized, 'evoked'.
```

Comment:
```suggestion
        obj_fun : string | func
            The objective function to be minimized.
```
This should be slightly more general, and I don't think `'evoked'` and taking the RMSE of simulated vs target dipoles are necessarily equivalent. How about the string label of `'dipole_rmse'`? Or what if we make this argument only for callables and not strings that translate to callables?